apache/iotdb
# [IOTDB-4835] Fix InnerSpaceCompaction may be blocked

See [IOTDB-4835](https://issues.apache.org/jira/browse/IOTDB-4835).

How to reproduce:

1. Set the compaction_target_file_size=100MB
2. Write one level-0 file whose size is greater than 100MB
3. Write several level-1 files whose size is smaller than 10MB
4. Run the IoTDB, the level-1 files will not be compacted, while they should be.

This PR fixes this bug by adjusting the rule of selecting SizeTieredCompactionTask.
